### PR TITLE
bin_to_octal

### PR DESCRIPTION
**Pull request by mohammadreza490**
Converts binary values to the octal equivalent.

### **Describe your change:**

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: conversions/bin_to_octal [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ValueError("Empty string was passed to the function")
+
+>>> bin_to_octal("a-1")
+ValueError("Non-binary value was passed to the function")
+
+"""
+
+
+def bin_to_octal(bin_string: str) -> str:
+    if not all("0" <= char <= "1" for char in bin_string):
```
Comment (cclauss) [inline conversions/bin_to_octal:20]:
```suggestion
    if not all(char in "01" for char in bin_string):
```

**Review by shellhub — CHANGES_REQUESTED** [review]
could you rename `bin_to_octal` to `bin_to_octal.py`

**Review by Charlie-xnxx — APPROVED** [review]

**Review by Charlie-xnxx — APPROVED** [review]

**Comment by shellhub** [comment]
My meaning is rename file `bin_to_octal` to `bin_to_octal.py` using below git command
``` bash
git mv bin_to_octal bin_to_octal.py
```

File: conversions/bin_to_octal [hunk 2]
```diff
@@ -0,0 +1,43 @@
+"""
+The function below will convert any binary string to the octal equivalent.
+
+>>> bin_to_octal("1111")
+'17'
+
```
Comment (shellhub) [inline conversions/bin_to_octal:6]:
```suggestion
```

File: conversions/bin_to_octal [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+>>> bin_to_octal("101010101010011")
+'52523'
+
```
Comment (shellhub) [inline conversions/bin_to_octal:9]:
```suggestion
```

File: conversions/bin_to_octal [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+>>> bin_to_octal("")
+ValueError("Empty string was passed to the function")
+
```
Comment (shellhub) [inline conversions/bin_to_octal:12]:
```suggestion
Traceback (most recent call last):
...
ValueError: Empty string was passed to the function
```

File: conversions/bin_to_octal [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+>>> bin_to_octal("a-1")
+ValueError("Non-binary value was passed to the function")
+
```
Comment (shellhub) [inline conversions/bin_to_octal:15]:
```suggestion
Traceback (most recent call last):
...
ValueError: Non-binary value was passed to the function
```

File: conversions/bin_to_octal [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+def bin_to_octal(bin_string: str) -> str:
+    if not all(char in "01" for char in bin_string):
+        raise ValueError("Non-binary value was passed to the function")
+    if not bin_string:
+        raise ValueError("Empty string was passed to the function")
+    oct_string = ""
+    while len(bin_string) % 3 != 0:
+        bin_string = "0" + bin_string
+    bin_string_in_3_list = [
+        bin_string[index : index + 3]
```
Comment (shellhub) [inline conversions/bin_to_octal:28]:
```suggestion
        bin_string[index: index + 3]
```

Comment (shellhub) [reply conversions/bin_to_octal:28]: delete whitespace before `:`

File: conversions/bin_to_octal (hunk 2 above)
Comment (mohammadreza490) [reply conversions/bin_to_octal:6]: There has to be some doctests in the code

**Comment by mohammadreza490** [comment]
> could you rename `bin_to_octal` to `bin_to_octal.py`

Done

**Comment by mohammadreza490** [comment]
> My meaning is rename file `bin_to_octal` to `bin_to_octal.py` using below git command
> 
> ```shell
> git mv bin_to_octal bin_to_octal.py
> ```

Done

**Review by shellhub — APPROVED** [review]

**Pull request merged**